spectral.yml

```diff
@@ -0,0 +1,141 @@
+extends: spectral:oas2
+rules:
+  operation-operationId: false
+  operation-tags: info
+  operation-2xx-response: error
+  paths-camelCase:
+    description: All YAML/JSON paths MUST follow camelCase
+    severity: warn
+    recommended: true
+    message: "{{property}} is not camelCase: {{error}}"
+    given: $.paths[*]~
+    then:
+      function: pattern
+      functionOptions:
+        # match: "/^(\/{1}(([{]?[a-z])[A-Za-z0-9]*[}]?)*)+$/" # - more generic one, allows /asasd{asdas}sadas pattern but also not closed braces
+        match: "^\/([a-z][a-zA-Z0-9]+)?(\/[a-z][a-zA-Z0-9]+|\/{[a-z][a-zA-Z0-9]+})*$" # doesn't allow /asasd{asdas}sadas pattern or not closed braces
+  definitions-camelCase-alphanumeric:
+    description: All YAML/JSON definitions MUST follow fields-camelCase and be ASCII alphanumeric characters or `_` or `$`.
+    severity: error
+    recommended: true
+    message: "{{property}} MUST follow camelCase and be ASCII alphanumeric characters or `_` or `$`."
+    given: $.definitions[*]~
+    then:
+      function: pattern
+      functionOptions:
+        match: "/^[a-z$_]{1}[A-Z09$_]*/"
+  properties-camelCase-alphanumeric:
+    description: All JSON Schema properties MUST follow fields-camelCase and be ASCII alphanumeric characters or `_` or `$`.
+    severity: error
+    recommended: true
+    message: "{{property}} MUST follow camelCase and be ASCII alphanumeric characters or `_` or `$`."
+    given: $.definitions..properties[*]~
+    then:
+      function: pattern
+      functionOptions:
+        match: "/^[a-z$_]{1}[A-Z09$_]*/"
+  protocol-https-only:
+    description: "ALL requests MUST go through `https` protocol only"
+    recommended: true
+    severity: error
+    type: "style"
+    message: "Schemes MUST be https and no other value is allowed."
+    given: $
+    then:
+      field: schemes
+      function: schema
+      functionOptions:
+        schema:
+          type: array
+          items:
+            type: string
+            enum: ["https"]
+          maxItems: 1
+  request-GET-no-body:
+    description: A `GET` request MUST NOT accept a `body` parameter
+    severity: error
+    given: $.paths..get.parameters..in
+    then:
+      function: pattern
+      functionOptions:
+        notMatch: "/^body$/"
+  response-success-hal: # schemes and/or produces
+    description: All success responses MUST be of media type `application/hal+json`
+    severity: error
+    given: $.paths..responses[?( @property >= 200 && @property < 300 )]
+    recommended: true
+    severity: error
+    type: "style"
+    message: "Response documents MUST follow application/hal+json: {{error}}"
+    then:
+      field: schema
+      function: schema
+      functionOptions:
+        schema:
+          $ref: ./hal.yaml

```

Comment:
Just `hal` should work, no?

Reply:
Supermodel - yes, spectral does not send the proper reference. That is what I want to do (supermodel) and this is a workaround.
`hal` - you man without `./`?
examples - i can do that.

Reply:
let me know when supermodel will be ready. As for README - i guess it would be enough to add it in the current one instead of creating a separate one?

Reply:
@jerzyn thanks! I hope to have the behavior of supermodel changed next week.  I will keep you posted. 

Regarding `hal` I’ve meant without the path prefix and file extension suffix. That should work according to the JSON Schema spec. Not sure if spectral handles it correctly. 

Regarding README, both will do, whichever you prefer.

Thanks again!

Reply:
spectral uses ajv for json-schema, but it is an older version for now (supports draft-04, newer one is coming soon)

Reply:
readme added in this PR: https://github.com/adidas/api-guidelines/pull/23